x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.ts

```diff
@@ -286,15 +297,54 @@ export const performBulkActionRoute = (
           savedObjectsClient,
         });
 
+        const query = body.query !== '' ? body.query : undefined;
+
+        if (body.action === BulkAction.edit) {
+          const { rules, errors } = await bulkEditRules({
+            rulesClient,
+            filter: query,
+            ids: body.ids,
+            actions: body.edit,
+            mlAuthz,
+          });
+
+          // migrate legacy rule actions
```

Comment:
legacy rule actions migration generates at least 2 find requests to ES, which are called every time on every rule, affecting overall performance of the API.
For example: with legacy migration 700 rules bulk edit takes ~ 30s, w/o only ~ 20s. so almost 50% slower

Is this something we need in this method? Can we get away from it here given we have it on other actions(specifically enable, when this migration looks like needed most)

Reply:
Hey, @elastic/security-detections-response-rules, any thoughts on this?

Reply:
Any updates on this? I believe Yara checked recently and determined that we can't get rid of the legacy migrations yet... but she would know best.

Reply:
I pinged @dhurley14 regarding that, but we won't remove the legacy actions migration call from bulk editing in this PR, so reviewing it shouldn't be a blocker.

Reply:
Mentioned below but yes we can't remove these. I think changing the ordering is fine so long as these tests are still passing. And don't look like they need updating.

https://github.com/elastic/kibana/blob/76e77c265b66fb031d3ba66bc7069affb797e981/x-pack/test/detection_engine_api_integration/security_and_spaces/group1/perform_bulk_action.ts#L432

Reply:
Sorry was on PTO - ++ what @dhurley14 said.